File: hyerang0125/0x05/boj_10828.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int x, n;
    string str;
    stack<int> s;

    cin >> n;
    for(int i=0; i<n; i++){
        cin >> str;
        if(str == "push"){
            cin >> x;
            s.push(x);
        }
        else if(str == "pop"){
            if(s.empty()) cout << "-1" << endl;
            else{
                cout << s.top() << endl;
                s.pop();
            }
        }
        else if(str == "size"){
            cout << s.size() << endl;
        }
        else if(str == "empty"){
            if(s.empty()) cout << "1" << endl;
            else cout << "0" << endl;
        }
        else if(str == "top"){
            if(s.empty()) cout << "-1" << endl;
            else cout << s.top() << endl;
        }
    }

    return 0;
}

File: minyoung/BOJ_10828.java
import java.util.StringTokenizer;
import java.util.Stack;
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main
{
	public static int N;
	public static void main (String[] args) throws java.lang.Exception
	{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st;
		StringBuilder sb = new StringBuilder();
		
		Stack <Integer> stack = new Stack<>();
		
		
		N = Integer.parseInt(br.readLine());
		
		for(int i=0;i<N;i++){
			st = new StringTokenizer(br.readLine());
			
			String str = st.nextToken();
			
			switch(str){
				case "push" :
					stack.push(Integer.parseInt(st.nextToken()));
					break;
				case "pop":
					if(!stack.empty()){
						sb.append(stack.pop());
					}
					else{
						sb.append("-1");
					}
					sb.append("\n");
					break;
				case "size":
					sb.append(stack.size());
					sb.append("\n");
					break;
				case "top":
					if(!stack.empty()){
						sb.append(stack.peek());
					}
					else{
						sb.append("-1");
					}
					sb.append("\n");
					break;
				case "empty":
					if(!stack.empty()){
						sb.append(0);
					}
					else{
						sb.append(1);
					}
					sb.append("\n");
					break;
			}
			
	
		}
		System.out.println(sb);
	}
}
